# Anwar abdi abdullahi                
## ID:C1210539                             
## Class:CA219

In [1]:
import numpy as np

In [3]:
import statistics as stats

In [5]:
data =np.array([18, 25, 12, 10, 30, 22, 15, 18, 20, 16, 19, 25, 21, 11, 14, 17, 28, 26, 23, 19])

In [7]:
mean_data =np.mean(data)
mean_data

19.45

### celceliska wakhtiga soo kabashada waa 19.45 maalmood, taasoo muujineysa in celcelis ahaan bukaannada ay ku soo kabtaan qiyaastii 19.5 maalmood.

In [10]:
meadian_data = np.median(data)
meadian_data

19.0

### (Median) waa qiimaha ku yaal bartamaha marka xogta la kala horreysiiyo. Xogtan, wakhtiga soo kabashada ee dhexdhexaadka ah waa 19 maalmood, taasoo macnaheedu yahay in kala bar bukaannada ay ka soo kabteen wax ka yar 19 maalmood, halka kala bar kale ay soo kabsadeen wax ka badan 19 maalmood.

In [13]:
mode_data = stats.mode(data)
mode_data

18

### Mode-ka waa qiimaha ugu badan ee soo noqnoqda ee xogta. Xogtan, qiimaha 18 ayaa ah kan ugu badan oo soo noqday 2 jeer, taasoo muujineysa in 18 maalmood uu yahay waqtiga soo kabashada ugu badan ee bukaannadu leeyihiin. 

In [16]:
range_data =np.max(data) - np.min(data)
range_data

20

### Range-ka (Farqiga u dhexeeya qiimaha ugu badan iyo kan ugu yar) waa cabbirka kala duwanaanshaha xogta. Xogta la bixiyay waxay leedahay kala duwanaansho dhan 20 maalmood, taasoo macnaheedu yahay in wakhtiga soo kabashada ugu gaaban uu yahay 10 maalmood, halka kan ugu dheer uu yahay 30 maalmood

In [19]:
variance_data = np.var(data, ddof=1)
std_deviation_data = np.std(data, ddof=1)
print("Variance:", variance_data)
print("Standard Deviation:", std_deviation_data)

Variance: 31.523684210526312
Standard Deviation: 5.614595640874445


### Kala duwanaanta 31.52 iyo heerka leexashada ee 5.61 waxay muujinayaan in wakhtiyada soo kabashada ay leeyihiin kala duwanaansho dhexdhexaad ah. Celcelis ahaan, wakhtiyada soo kabashada ayaa ku kala duwan qiyaastii 5.61 maalmood marka loo eego celceliska.

In [22]:
percentile_25 = np.percentile(data, 25)
percentile_50 = np.percentile(data, 50) 
percentile_75 = np.percentile(data, 75)

print("25th Percentile:", percentile_25)
print("50th Percentile (Median):", percentile_50)
print("75th Percentile:", percentile_75)

25th Percentile: 15.75
50th Percentile (Median): 19.0
75th Percentile: 23.5


### Boqolleyda 25-aad waxay kuu sheegaysaa in 25% ee wakhtiyada soo kabashada ay yihiin 15.75 maalmood ama ka yar, halka boqolleyda 75-aad ay tilmaamayso in 75% ee wakhtiyada soo kabashada ay yihiin 23.5 maalmood ama ka yar. Boqolleyda 50-aad waa dhex-dhexaadka xogta.

In [25]:
percentile_25 = np.percentile(data, 25)
percentile_75 = np.percentile(data, 75)

In [27]:
IQR = percentile_75 - percentile_25
print("Interquartile Range (IQR):", IQR)

Interquartile Range (IQR): 7.75


### IQR-ka waa 7.75 maalmood. Taas micnaheedu waa in 50% ee dhexe ee wakhtiyada soo kabashada ay ku jiraan farqiga 7.75 maalmood.

# Warbixin Kooban: Falanqaynta Wakhtiyada Soo Kabashada ee Bukanadda
Warbixintan waxay falanqaysaa xogta wakhtiyada soo kabashada (maalmo) iyadoo la adeegsanayo falanqaynta tirakoobka. Xogta la falanqeeyay waxaa ka mid ah celcelis (mean), dhexdhexaad (median), mode (qiimaha ugu badan ee soo noqnoqda), kala duwanaanshaha (range), farqiga (variance), heerka leexashada (standard deviation), boqolleeyada (percentiles), iyo cabbirka kala-duwanaanshaha dhexe (IQR).

### Soo Koobid Warbixin:
1. Celcelis (Mean): Wakhtiga celceliska ah ee soo kabashada waa 19.45 maalmood, taasoo muujineysa in bukaanada inta badan ay soo kabsadaan qiyaastii 19.5 maalmood.

2. Dhexdhexaad (Median): Wakhtiga soo kabashada ee bartamaha waa 19 maalmood, taasoo macnaheedu yahay kala bar bukaannada waxay soo kabteen wax ka yar ama la eg 19 maalmood, halka kala barka kale ay ka badan yihiin.

3. Mode: Qiimaha soo noqnoqda ugu badan ee xogta waa 18 maalmood, taasoo muujineysa in bukaannada badankood ay ku soo kabteen 18 maalmood.

4. Range: Kala duwanaanshaha wakhtiyada soo kabashada waa 20 maalmood, laga bilaabo ugu yaraan 10 maalmood ilaa ugu badnaan 30 maalmood. Tani waxay muujineysaa kala duwanaansho weyn oo ku saabsan wakhtiyada soo kabashada.

5. Farqiga (Variance) iyo Heerka Leexashada (Standard Deviation):

Farqiga waa 31.52, halka heerka leexashada uu yahay 5.61 maalmood, taasoo muujineysa in wakhtiyada soo kabashada ay celcelis ahaan ku kala duwan yihiin qiyaastii 5.6 maalmood marka loo eego celceliska.

6. Boqolleeyada (Percentiles):

  25% ee bukaannada waxay soo kabteen wax ka yar 15.75 maalmood.
  50% (median) waxay soo kabteen 19 maalmood ama ka yar.
  75% waxay soo kabteen 23.5 maalmood ama ka yar.
  
7. IQR (Interquartile Range): IQR waa 7.75 maalmood, taasoo macnaheedu yahay in kala badhka dhexe ee    bukaannada ay ku soo kabteen farqiga 7.75 maalmood ee u dhexeeya boqolleeyada 25-aad iyo 75-aad.

### Soo Koobid Guud:
Wakhtiyada soo kabashada ee bukaanadu waxay u dhexeeyaan 19.45 maalmood oo celcelis ah, iyada oo kala duwanaansho dhexdhexaad ah (heerka leexashada waa 5.61 maalmood). Inta badan bukaannada waxay ku soo kabtaan wakhtiyo u dhexeeya 15.75 ilaa 23.5 maalmood.

Warbixintani waxay si buuxda u sharxaysaa qaabka wakhtiyada soo kabashada ee bukaanada, iyadoo la adeegsanayo falanqaynta tirakoobka si loo fahmo habka soo kabashada.